metrics/web/test/base/presentation/widgets/clearable_text_form_field_test.dart

```diff
@@ -0,0 +1,197 @@
+      (tester) async {
+        await tester.pumpWidget(
+          const _ClearableTextFormFieldTestbed(label: label, controller: null),
+        );
+
+        expect(tester.takeException(), isAssertionError);
+      },
+    );
+
+    testWidgets(
+      "applies the given label",
+      (WidgetTester tester) async {
+        final controller = TextEditingController();
+
+        await tester.pumpWidget(
+          _ClearableTextFormFieldTestbed(label: label, controller: controller),
+        );
+
+        expect(find.text(label), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "does not display a clear icon if an input is empty",
+      (WidgetTester tester) async {
+        final controller = TextEditingController();
+
+        await tester.pumpWidget(
+          _ClearableTextFormFieldTestbed(label: label, controller: controller),
+        );
+
+        expect(
+          find.descendant(
+            of: find.byType(IconButton),
+            matching: find.byIcon(Icons.close),
+          ),
+          findsNothing,
+        );
+      },
+    );
+
+    testWidgets(
+      "displays a clear icon if an input is not empty",
+      (WidgetTester tester) async {
+        final controller = TextEditingController(text: 'text');
+
+        await tester.pumpWidget(
+          _ClearableTextFormFieldTestbed(label: label, controller: controller),
+        );
+
+        expect(
+          find.descendant(
+            of: find.byType(IconButton),
+            matching: find.byIcon(Icons.close),
+          ),
+          findsOneWidget,
+        );
+      },
+    );
+
+    testWidgets(
+      "validator callback is called on the form validation",
+      (WidgetTester tester) async {
+        bool _callbackIsCalled = false;
+        final _formKey = GlobalKey<FormState>();
+
+        await tester.pumpWidget(
+          _ClearableTextFormFieldTestbed(
+            formKey: _formKey,
+            label: label,
+            controller: TextEditingController(text: 'text'),
+            validator: (String value) {
+              _callbackIsCalled = true;
+              return null;
+            },
+          ),
+        );
+
+        _formKey.currentState.validate();
+
+        expect(_callbackIsCalled, isTrue);
+      },
+    );
+
+    testWidgets(
+      "applies the given border",
+      (WidgetTester tester) async {
+        final controller = TextEditingController(text: 'text');
+        const border = OutlineInputBorder(
+          borderSide: BorderSide(color: Colors.red),
+        );
+
+        await tester.pumpWidget(
+          _ClearableTextFormFieldTestbed(
+            label: label,
+            controller: controller,
+            border: border,
+          ),
+        );
+
+        final widget = tester.widget<ClearableTextFormField>(
+          find.byType(ClearableTextFormField),
+        );
+
+        expect(widget.border, border);
+      },
+    );
+
+    testWidgets(
+      "clears the text field on tap on clear icon",
```

Comment:
```suggestion
      "clears an input on tap on a clear icon",
```
to make it consistent with other text descriptions